packages/ui/src/partials/w3m-account-button/index.ts

```diff
@@ -1,16 +1,78 @@
-import { ClientCtrl } from '@web3modal/core'
+import { AccountCtrl, ConnectModalCtrl, RouterCtrl } from '@web3modal/core'
 import { html } from 'lit'
-import { customElement } from 'lit/decorators.js'
+import { customElement, state } from 'lit/decorators.js'
+import '../../components/w3m-text'
+import '../../components/w3m-zorb-ens-image'
+
+import { CHEVRON, ETH_LOGO } from '../../utils/Svgs'
 import { global } from '../../utils/Theme'
 import ThemedElement from '../../utils/ThemedElement'
+import { formatAddress, roundBalance } from '../../utils/UiHelpers'
+import styles, { dynamicStyles } from './styles'
 
 @customElement('w3m-account-button')
 export class W3mAccountButton extends ThemedElement {
-  public static styles = [global]
+  public static styles = [global, styles]
+
+  // -- state & properties ------------------------------------------- //
+  @state() private address = ''
+  @state() private balance = ''
+  @state() private ensAvatar = ''
+
+  // -- lifecycle ---------------------------------------------------- //
+  public constructor() {
+    super()
+    this.subscribeAccountChanges()
+  }
+
+  // -- private ------------------------------------------------------ //
+
+  private unsubscribe?: () => void = undefined
+
+  private subscribeAccountChanges() {
+    this.address = AccountCtrl.state.address
+    this.balance = AccountCtrl.state.balance
+    this.ensAvatar = AccountCtrl.state.ensAvatar
+
+    this.unsubscribe = AccountCtrl.subscribe(() => {
+      this.address = AccountCtrl.state.address
+      this.ensAvatar = AccountCtrl.state.ensAvatar
+      this.balance = AccountCtrl.state.balance
+    })
+
+    return () => this.unsubscribe
```

Comment:
Why is this return needed here? You set `this.unsubscribe` correctly, but never unsubscribe actually. need to add

```ts
public disconnectedCallback() {
    this.unsubscribe?.()
  }
```